Src/FluentAssertions/Specialized/GenericAsyncFunctionAssertions.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Specialized
+{
+    public class GenericAsyncFunctionAssertions<TResult> : AsyncFunctionAssertions
+    {
+        private readonly Func<Task<TResult>> subject;
+        private readonly ITimer timer;
+
+        public GenericAsyncFunctionAssertions(Func<Task<TResult>> subject, IExtractExceptions extractor, ITimer timer) : base(
+            subject, extractor)
+        {
+            this.subject = subject;
+            this.timer = timer;
+        }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Task{T}"/> will complete within specified time.
+        /// </summary>
+        /// <param name="timeSpan">The allowed time span for the operation.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndWhichConstraint<GenericAsyncFunctionAssertions<TResult>, Task<TResult>> CompleteWithin(
```

Comment:
Could we return `TResult` instead of `Task<TResult>`?
That would remove the need in tests to call `.Result`.